daffodil-core/src/main/scala/org/apache/daffodil/runtime2/GeneratedCodeCompiler.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.runtime2
+
+import org.apache.commons.io.FileUtils
+import org.apache.daffodil.compiler.ProcessorFactory
+import org.apache.daffodil.dsom.SchemaDefinitionError
+import org.apache.daffodil.runtime2.generators.CodeGeneratorState
+import org.apache.daffodil.util.Misc
+import os.Pipe
+
+class GeneratedCodeCompiler(pf: ProcessorFactory) {
```

Comment:
I would expect all this logic to be in the runtime2 subproject. So runtime is essentially made up of jar that generates C code and compiles it with the static library. Perhaps this library could even live inside the jar and extracted for the during compilation process? So there would be no need to actually distribute the static lib in the zip/tar/rpm since it's all part of the jar?

Reply:
Agreed, I'd like to hear others thoughts. I don't think I have a strong preference either way.

I'm wondering if there are may IDE issues? We can definitely support this C compiler plugin in sbt, but if people use other IDEs, maybe they would have a harder time building everything since this is sort of Java focused project. I assuming eclipse and others can deal with mixing Java and C, but can it package a static lib in a jar? We'd robably want to update our eclipse configs so this "just works" for people wanting to use SBT. Other IDE's to consider as well.